pkg/adapter/awsperformanceinsightssource/adapter.go

```diff
@@ -150,11 +159,16 @@ func (a *adapter) PollMetrics(priorTime time.Time, currentTime time.Time) {
 	for _, d := range rm.MetricList {
 		for _, metric := range d.DataPoints {
 			if metric.Value != nil {
+				e := &event{
+					Metric: *d.Key.Metric,
+					Value:  *metric.Value,
```

Comment:
Curious why we trim things like that instead of just sending the whole `metric` struct?

Reply:
because it will emit tons of events full of timestamps with no values if we do not filter and trim